python/ray/data/_internal/datasource/webdataset_datasource.py

```diff
@@ -362,4 +362,17 @@ def get_tar_file_iterator():
         for sample in samples:
             if self.decoder is not None:
                 sample = _apply_list(self.decoder, sample, default=_default_decoder)
+            if self.expand_json:
```

Comment:
will this always have `json` field?

Reply:
Yes. For example, a sample of the JSON column in WebDataset is as follows:
{
    "file_name": "00001.jpg",
    "height":50
}, Ultimately, it will be formed into something like this:
|  file_name   | height  |json|
|  ----        | ----    |----|
| 00001.jpg    | 50      |{"file_name": "00001.jpg","height":50}|